src/lighthouse.rs

```diff
@@ -138,16 +141,35 @@ fn quorum_compute(
     // Sort by replica ID to get a consistent ordering across runs.
     candidate_participants.sort_by_key(|p| p.replica_id.clone());
 
+    let shrink_only = healthy_participants
```

Comment:
do we expect to have only some members have `shrink_only=True` and some `shrink_only=False`? I noticed we take `any` but should all the members be consistent?

Reply:
That is an expected case -- in LocalSGD we could have members join at any time and be in the wrong spot in the training loop. Members in the right location should already be in the quorum. In that case, the members not participating will get an exception and it's up to the caller to handle this correctly